pkg/controllermanager/apis/config/types.go

```diff
@@ -180,6 +180,8 @@ type ShootMaintenanceControllerConfiguration struct {
 	ConcurrentSyncs int
 	// EnableShootControlPlaneRestarter configures whether adequate pods of the shoot control plane are restarted during maintenance.
 	EnableShootControlPlaneRestarter *bool
+	// EnableShootCoreDNSRestarter configures whether CoreDNS deployment to be  restarted during maintenance.
+	EnableShootCoreDNSRestarter *bool
```

Comment:
How about naming this `EnableShootCoreAddonRestarter` which leaves more room for future use-cases?